### PR TITLE
Bump prettier to 3.3.3 and reformat files

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Bump prettier to 3.3.3 and run against all files
- See linked code 

## Discussion

- Do we need prettier as an npm dev dependency? Do people's IDEs use that if they have it configured (or does that functionality use npx)?

## Future Work

- Create a job to automatically bump prettier and run over all files

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: applications/browser-extension/src/analysis/analysisVisitors/varAnalysis/varAnalysis.ts [hunk 1]
```diff
@@ -225,7 +225,7 @@ function setVarsFromSchema({
       const nodePath = [...parentPath, key];
 
       const existence =
-        existenceOverride ?? required?.includes(key)
+        (existenceOverride ?? required?.includes(key))
```
Comment (twschiller) [inline applications/browser-extension/src/analysis/analysisVisitors/varAnalysis/varAnalysis.ts:228]: 💅

File: package.json [hunk 2]
```diff
@@ -25,9 +25,9 @@
     "npm": "10.5.0"
   },
   "devDependencies": {
-    "knip": "^5.36.7",
```
Comment (twschiller) [inline package.json:28]: Let's keep these sorted

File: .prettierignore [hunk 3]
```diff
@@ -4,7 +4,7 @@ coverage
 
 # ignore libraries
 node_modules
-public/mockServiceWorker.js
+**/public/mockServiceWorker.js
```
Comment (twschiller) [inline .prettierignore:7]: @fungairino @grahamlangford we'll want to consider workspace-specific `prettierignore` and `gitignore` files

**Comment by grahamlangford** [comment]
> Do we need prettier as an npm dev dependency? Do people's IDEs use that if they have it configured (or does that functionality use npx)?

Exactly. Remember when we had to commit, add, commit in order to satisfy the prettier step in pre-commit? Making prettier a devDependency ensured that pre-commit and local format-on-save agreed

File: applications/browser-extension/src/analysis/analysisVisitors/varAnalysis/varAnalysis.ts (hunk 1 above)
Comment (grahamlangford) [reply applications/browser-extension/src/analysis/analysisVisitors/varAnalysis/varAnalysis.ts:228]: Big fan of this prettier change

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**